2-mandatory/3-financial-times.js

```diff
@@ -23,15 +31,17 @@ function titleWithFewestWords(allArticleTitles) {
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
 function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+    return allArticleTitles.filter(x=>x.match(/\d+/g))
 }
 
 /*
     The Financial Times wants to understand what the average number of characters in an article title is.
     Implement the function below to return this number - rounded to the nearest integer.
 */
 function averageNumberOfCharacters(allArticleTitles) {
-    // TODO
+
+    let sumOfChar = allArticleTitles.map(x=>x.length)
+    return Math.round(sumOfChar.reduce(function(a, b) { return a + b; }, 0)/allArticleTitles.length)
```

Comment:
Nice! with `reduce` you have access to the element under the reference `b`. `a` is the accumulator.  You could also check the length there. and get rid of line 43.
`.reduce(function(a, b) { return (a + b.length) },0)`

Reply:
thank you, it working with new approach